k8s/eks/README.md

```diff
@@ -0,0 +1,284 @@
+
+# or in this case since the images are moved to a public repo:
+
+aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin public.ecr.aws/n6b0k8i7
+```
+
+**5. eksctl**
+
+Official installation guide:
+
+https://docs.aws.amazon.com/eks/latest/userguide/eksctl.html
+
+```
+curl --silent --location "https://github.com/weaveworks/eksctl/releases/latest/download/eksctl_$(uname -s)_amd64.tar.gz" | tar xz -C /tmp
+sudo mv /tmp/eksctl /usr/local/bin
+eksctl version
+```
+
+
+## Step by step installation guide
+
+1. Pull the infra repo from here:
+
+git clone https://github.com/testground/infra.git
+
+and the testground repo from here:
+
+git clone https://github.com/testground/testground
+
+2. You need to populate the `.env` file with the needed parameters before creating the cluster. It is located in `infra/k8s/eks/`
+
+3. Run the installation script `./testground_install.sh`; note that it requires the `.env` file to be populated in order to run properly. In case of any missing variables, the setup will fail and you will get an error message
+
+The script explained in short:
+
+- Takes approx. 15-20 minutes to provision a cluster with 2 worker nodegroups, along with all other necessary resources and cluster workloads
+- All installation steps have been divided into functions, located in the `eks/bash/functions.sh` script
+- The script calls functions one by one
+- Output of every step is logged into a master log file that can be retrieved and reviewed, or even tailed in real time
+- It can also be used to uninstall a provisioned cluster; one of the first steps in the script is checking for existing resources
+- The `functions.sh` file relies on the content of `eks/yaml` - all cluster resources are located inside this folder
+- If you need to scale the worker nodegroups, or introduce any other changes, you are able to edit the `functions.sh` script
+- Once the cluster is created, eksctl will automatically switch the context to the new cluster. If you create another cluster, eksctl will once again switch the context to the newest cluster. If you wish to switch to another cluster, issue:
+
+```
+kubectl config get-contexts
+
+# currently selected cluster will be mark with an asterisk (*)
+# then switch contexts to the desired cluster
+
+kubectl config use-context $USERNAME@$CLUSTER_NAME.$AWS_REGION.eksctl.io
+```
+
+4. Create the `.env.toml` file inside the `testground` folder and populate it (explained here as well https://github.com/testground/docs/blob/master/getting-started.md#configuration-envtoml):
+
+**Note: The endpoint refers to the `testground-daemon` service, so depending on your setup, this could be, for example, a Load Balancer fronting the kubernetes cluster and forwarding proper requests to the `tg-daemon` service, or a simple port forward to your local workstation.**
```

Comment:
Could you please clarify this?

Reply:
When I look at the output of `testground_install` script, how do I find the value I should use for the `endpoint`? Could we mark it in a way to would make it obvious?

Reply:
I believe the script doesn't output that in this iteration, but it can be added.
Right now it is explained in the README under step `6.` how to set up the LB.

Reply:
We should have that information readily available. If we could get https://github.com/testground/infra/pull/78#discussion_r984576809 though, it'd be perfect.

Reply:
Yeah I agree, we will have to modify the script so it outputs this for you.
It is now missing because we added LB support late last week and didn't work on the script as well.

Reply:
The script now outputs the .env.toml once it finishes, so the users only need to copy and paste it in their testground/.env.toml file.
It also includes the public DNS of the LB.